torch/testing/_internal/common_distributed.py

```diff
@@ -1456,12 +1468,18 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
 @contextmanager
 def _dynamo_dist_per_rank_init(
-    rank, world_size, backend="nccl", init_pg=True, fake_pg=False
+    rank, world_size, backend=None, init_pg=True, fake_pg=False
 ):
     # To avoid multiple inheritance from _dynamo.test_case.TestCase and MultiProcessTestCase,
     # Just manually implement the most important part of the dynamo behavior to reset/clear.
     if not fake_pg:
         torch.accelerator.set_device_index(rank)
+
+    if backend is None:
+        backend = c10d.get_default_backend_for_device(
+            torch.accelerator.current_accelerator().type
+        )
```

Comment:
```suggestion
        device_type = acc.type if (acc := torch.accelerator.current_accelerator()) else "cpu"
        backend = c10d.get_default_backend_for_device(device_type)
```
Otherwise, `torch.accelerator.current_accelerator()` will return `None` if no accelerator detected.

Reply:
Updated, thanks!